File: Difficulty: Easy/Count the Zeros/count-the-zeros.py
#User function Template for python3

class Solution:
    def countZeroes(self, arr):
        # code here
        ans = -1
        start = 0
        end = len(arr) - 1

        while (start <= end):
            mid = (start + end) // 2

            if arr[mid] == 1:
                start = mid + 1
            else:
                ans = mid
                end = mid - 1

        return len(arr) - ans

File: Difficulty: Easy/Count the Zeros/count-the-zeros.cpp
class Solution {
  public:
    int countZeroes(vector<int> &arr) {
        int n = arr.size();
        int left = 0, right = n - 1;
        int firstZero = -1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (arr[mid] == 0) {
                firstZero = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        if (firstZero == -1) return 0;

        return n - firstZero;
    }
};